src/main/java/com/squareup/javapoet/MemberRef.java

```diff
@@ -0,0 +1,138 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.javapoet;
+
+import java.io.IOException;
+import java.lang.reflect.Member;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.List;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.type.TypeMirror;
+
+@SuppressWarnings("javadoc")
+public final class MemberRef {
+
+  public static MemberRef get(Enum<?> constant) {
+    Util.checkNotNull(constant, "constant == null");
+    return new MemberRef(TypeName.get(constant.getDeclaringClass()), constant.name());
+  }
+
+  public static MemberRef get(Member member, Type... types) {
+    Util.checkNotNull(member, "member == null");
+    Util.checkNotNull(types, "types == null");
+    Util.checkState(java.lang.reflect.Modifier.isStatic(member.getModifiers()), "not static");
+    TypeName type = TypeName.get(member.getDeclaringClass());
+    return new MemberRef(type, member.getName(), TypeName.list(types));
+  }
+
+  public static MemberRef get(Member member, Object someObject, Type... types) {
+    Util.checkNotNull(member, "member == null");
+    Util.checkNotNull(someObject, "someObject == null");
+    Util.checkNotNull(types, "types == null");
+    TypeName type = TypeName.get(member.getDeclaringClass());
+    return new MemberRef(type, member.getName(), someObject, TypeName.list(types));
+  }
+
+  public static MemberRef get(Element element, TypeMirror... types) {
+    Util.checkNotNull(element, "element == null");
+    Util.checkNotNull(types, "types == null");
+    Util.checkState(element.getModifiers().contains(Modifier.STATIC), "element not STATIC");
+    TypeName type = TypeName.get(element.getEnclosingElement().asType());
+    String name = element.getSimpleName().toString();
+    return new MemberRef(type, name, TypeName.list(types));
+  }
+
+  public static MemberRef get(Element element, Object someObject, TypeMirror... types) {
+    Util.checkNotNull(element, "element == null");
+    Util.checkNotNull(someObject, "someObject == null");
+    Util.checkNotNull(types, "types == null");
+    TypeName type = TypeName.get(element.getEnclosingElement().asType());
+    String name = element.getSimpleName().toString();
+    return new MemberRef(type, name, someObject, TypeName.list(types));
+  }
+
+  public final TypeName type;
+  public final List<TypeName> arguments;
+  public final String name;
+  public final Object someObject;

```

Comment:
What is the line, you're referring to?

If it's line 73 `public final Object someObject;` then it's the call site, the target, the instance that the method is called on.